utils/reflector/src/main/java/org/robolectric/util/reflector/UnsafeAccess.java

```diff
@@ -62,7 +62,7 @@ public <T> Class<?> defineClass(Class<T> iClass, String reflectorClassName, byte
                 iClass.getClassLoader(),
                 null);
       } catch (IllegalAccessException | InvocationTargetException e) {
-        throw new AssertionError(e);
+        throw new LinkageError(e.getMessage(), e);
```

Comment:
`LinkageError` is more clear than `AssertionError`, and it's safe to converting them for Robolectric's internal test code. But I wonder whether it breaks user's code if they use try-catch to catch `AssertionError` after this PR changes it to `LinkageError`. Both `LinkageError` and `AssertionError` inherit from `Error` and they don't have directly inheritance relationship. @hoisie What do you think about it?

Reply:
Oh, we can suppress it. @hellosagar we can keep fixes for Test code, but suppress this warning for released code.

Reply:
Suppressed in release code
- https://github.com/robolectric/robolectric/pull/7400/files#diff-d073a1c9ddfeede22b2a22843b117202e35c801471da6cc6c821c79d5123a90aR23
- https://github.com/robolectric/robolectric/pull/7400/files#diff-0a35332fe687da483e340cf36efeea2ffdc86871245c84d1fbcf847852c55094